version/string.go

```diff
@@ -4,37 +4,54 @@
 package version
 
 import (
+	"encoding/json"
 	"fmt"
 	"runtime"
+	"strconv"
 	"strings"
 )
 
-var (
-	// String is displayed when CLI arg --version is used
-	String string
+// GitCommit is set in the build script at compile time
+var GitCommit string
 
-	// GitCommit is set in the build script at compile time
-	GitCommit string
-)
+// Versions contains the versions relevant to a build of avalanchego. In
+// addition to supporting construction of the strings displayed by
+// --version and --json-version, it can be used to unmarshal the output
+// of --json-version.
+type Versions struct {
+	Application string `json:"application"`
+	Database    string `json:"database"`
+	RPCChainVM  string `json:"rpcchainvm"`
+	Commit      string `json:"commit"`
+	Go          string `json:"go"`
+}
 
-func init() {
-	format := "%s [database=%s, rpcchainvm=%d"
-	args := []interface{}{
-		CurrentApp,
-		CurrentDatabase,
-		RPCChainVMProtocol,
+func GetVersions() *Versions {
+	versions := &Versions{
+		Application: CurrentApp.String(),
+		Database:    CurrentDatabase.String(),
+		RPCChainVM:  strconv.FormatUint(uint64(RPCChainVMProtocol), 10),
+		Go:          strings.TrimPrefix(runtime.Version(), "go"),
 	}
 	if GitCommit != "" {
-		format += ", commit=%s"
-		args = append(args, GitCommit)
+		versions.Commit = GitCommit
 	}
+	return versions
+}
 
-	// add golang version
-	goVersion := runtime.Version()
-	goVersionNumber := strings.TrimPrefix(goVersion, "go")
-	format += ", go=%s"
-	args = append(args, goVersionNumber)
+func (v *Versions) String() string {
+	// This format maintains consistency with previous --version output
+	versionString := fmt.Sprintf("%s [database=%s, rpcchainvm=%s, ", v.Application, v.Database, v.RPCChainVM)
+	if len(v.Commit) > 0 {
+		versionString += fmt.Sprintf("commit=%s, ", v.Commit)
+	}
+	return versionString + fmt.Sprintf("go=%s]", v.Go)
+}
 
-	format += "]\n"
-	String = fmt.Sprintf(format, args...)
+func (v *Versions) JSON() string {
+	jsonBytes, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		panic(err)
+	}
+	return string(jsonBytes)
 }
```

Comment:
This seems a bit non-standard. I feel like the caller should call `json.MarshalIndent` on their own terms and handle the error however they want.